### PR TITLE
[SPARK-13457][SQL] Removes DataFrame RDD operations

### PR DESCRIPTION
**Pull request by liancheng**
## What changes were proposed in this pull request?

This is another try of PR #11323.

This PR removes DataFrame RDD operations except for `foreach` and `foreachPartitions` (they are actions rather than transformations). Original calls are now replaced by calls to methods of `DataFrame.rdd`.

PR #11323 was reverted because it introduced a regression: both `DataFrame.foreach` and `DataFrame.foreachPartitions` wrap underlying RDD operations with `withNewExecutionId` to track Spark jobs. But they are removed in #11323.
## How was the this patch tested?

No extra tests are added. Existing tests should do the work.

### CODE REVIEW

**Comment by liancheng** [comment]
Merged to master.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 1]
```diff
@@ -1427,30 +1427,6 @@ class DataFrame private[sql](
   def transform[U](t: DataFrame => DataFrame): DataFrame = t(this)
 
   /**
-   * Returns a new RDD by applying a function to all rows of this DataFrame.
-   * @group rdd
-   * @since 1.3.0
-   */
-  def map[R: ClassTag](f: Row => R): RDD[R] = rdd.map(f)
```
Comment (JoshRosen) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: Surprised this didn't trigger MiMA failures. Know why?

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala [hunk 2]
```diff
@@ -85,7 +85,8 @@ final class RegressionEvaluator @Since("1.4.0") (@Since("1.4.0") override val ui
 
     val predictionAndLabels = dataset
       .select(col($(predictionCol)).cast(DoubleType), col($(labelCol)).cast(DoubleType))
-      .map { case Row(prediction: Double, label: Double) =>
+      .rdd.
```
Comment (cloud-fan) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala]: the code style is weird here...

Comment (liancheng) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala]: Thanks, will fix this in future PRs.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala (hunk 1 above)
Comment (liancheng) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: Good question... Checked the Jenkins MiMA section of the build log of this PR, didn't see any lines related to DataFrame.

**Pull request closed without merging**